### PR TITLE
Bicaws 2213 lock record when viewed

### PR DESCRIPTION
**Pull request by C-gyorfi**
### Changes
image

### CODE REVIEW

File: src/services/fetchAndTryLockCourtCase.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+): Promise<fetchAndTryLockCourtCaseResult> => {
+  const ok = await tryToLockCourtCase(dataSource, courtCaseId, currentUser.username)
+  if (!ok) {
+    return {
+      error: true
+    }
+  }
+
+  const courtCase = await getCourtCase(dataSource, courtCaseId, currentUser.visibleForces)
+  if (!courtCase) {
```
Comment (C-gyorfi) [inline src/services/fetchAndTryLockCourtCase.ts:23]: Is this a case of an error? `getCourtCase` would return null if the court case doesn't exist so I think we need to handle as not found

File: test/services/tryToLockCourtCase.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    const result = await tryToLockCourtCase(dataSource, existingCourtCase.errorId, "Bichard01")
+    expect(isError(result)).toBe(false)
+
+    const record = await dataSource.getRepository(CourtCase).findOne({ where: { errorId: 0 } })
+    const actualCourtCase = record as CourtCase
+    expect(actualCourtCase.errorLockedById).toStrictEqual(anotherUser)
+    expect(actualCourtCase.triggerLockedById).toStrictEqual(anotherUser)
+  })
+
+  // These tests are out of scope until separate case and trigger locking is implemented
```
Comment (C-gyorfi) [inline test/services/tryToLockCourtCase.integration.test.ts:76]: I think we want to remove these comments

File: test/services/fetchAndTryLockCourtCase.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    await fetchAndTryLockCourtCase(
+      { username: "bichard01", visibleForces: ["36"] } as unknown as User,
+      unlockedCourtCase.error_id,
+      dataSource
+    )
+
+    const updatedCourtCase: CourtCase = await getCourtCase(dataSource, unlockedCourtCase.error_id, ["36"])
+    insertedCourtCase.errorLockedById = "bichard01"
+    insertedCourtCase.triggerLockedById = "bichard01"
```
Comment (C-gyorfi) [inline test/services/fetchAndTryLockCourtCase.integration.test.ts:72]: This test is a bit confusing, we could remove `insertedCourtCase` and do 
```
expect(updatedCourtCase.errorLockedById).toStrictEqual("bichard01")
expect(updatedCourtCase.triggerLockedById).toStrictEqual("bichard01")
```
to run less queries

File: src/services/fetchAndTryLockCourtCase.ts (hunk 1 above)
Comment (tomvaughan77) [reply src/services/fetchAndTryLockCourtCase.ts:23]: If you look at L23 in the case details page (`index.tsx`), this gets handled as a `notFound` already 😄

**Review by tomvaughan77 — APPROVED** [review]

**Pull request merged**